python/taichi/linalg/taichi_cg.py

```diff
@@ -0,0 +1,99 @@
+from math import sqrt
+
+import taichi as ti
+
+
+@ti.data_oriented
+class LinearOperator:
+    def __init__(self, matvec_kernel):
+        self._matvec = matvec_kernel
+
+    def matvec(self, x, Ax):
+        assert x.shape == Ax.shape, "Dimension mismatch x.shape != Ax.shape."
+        self._matvec(x, Ax)
+
+
+def taichi_cg_solver(A, b, x, tol=1e-6, maxiter=5000, quiet=True):
+    assert b.dtype == x.dtype, "Dtype mismatch b.dtype != x.dtype."
```

Comment:
I think you should raise an exception here :)

Reply:
Thank you for the suggestion!